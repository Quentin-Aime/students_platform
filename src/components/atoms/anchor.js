import React from "react";
import styled from "styled-components";
import { color } from "../../assets/const";
import AnchorLink from "react-anchor-link-smooth-scroll";

export const Anchor = ({ label, url, style }) => {
  return (
    <>
      <Link href={`#${url}`} style={style}>
        {label}
      </Link>
    </>
  );
};

const Link = styled(AnchorLink)`
  text-decoration: none;
  text-transform: uppercase;
  color: ${color.primary};
  +a {
    margin-left: 20px;
  }
  &:hover {
    color: ${color.secondary};
  }
`;
