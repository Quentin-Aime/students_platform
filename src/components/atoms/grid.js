import React from "react";
import styled from "styled-components";

export const Grid = ({ children }) => {
  return <Container>{children}</Container>;
};

const Container = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;
