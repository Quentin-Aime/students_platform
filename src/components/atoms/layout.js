import React from "react";
import styled from "styled-components";

export const Layout = ({ children }) => {
  return <Wrapper>{children}</Wrapper>;
};

const Wrapper = styled.div`
  width: 100%;
  padding: 3rem;
`;
