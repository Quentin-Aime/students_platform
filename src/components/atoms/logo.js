import React from "react";
import styled from "styled-components";

export const Logo = () => {
  return (
    <Container>
      <svg
        width="64"
        height="64"
        viewBox="0 0 64 64"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20.9195 27.7151C31.7195 7.71506 44.4195 6.71506 49.4195 8.71506C68.4195 8.71506 64.9195 27.7151 59.9195 45.7151C55.9195 60.1151 39.5861 66.7151 31.9195 62.7151C24.2528 58.7151 10.1195 47.7151 20.9195 27.7151Z"
          fill="#F9D2A4"
        />
        <path
          d="M20 25C20 24.4067 19.8241 23.8266 19.4944 23.3333C19.1648 22.8399 18.6962 22.4554 18.1481 22.2284C17.5999 22.0013 16.9967 21.9419 16.4147 22.0576C15.8328 22.1734 15.2982 22.4591 14.8787 22.8787C14.4591 23.2982 14.1734 23.8328 14.0576 24.4147C13.9419 24.9967 14.0013 25.5999 14.2284 26.1481C14.4554 26.6962 14.8399 27.1648 15.3333 27.4944C15.8266 27.8241 16.4067 28 17 28C17.7954 27.9991 18.5579 27.6828 19.1203 27.1203C19.6828 26.5579 19.9991 25.7954 20 25ZM16 25C16 24.8022 16.0586 24.6089 16.1685 24.4444C16.2784 24.28 16.4346 24.1518 16.6173 24.0761C16.8 24.0004 17.0011 23.9806 17.1951 24.0192C17.3891 24.0578 17.5673 24.153 17.7071 24.2929C17.847 24.4327 17.9422 24.6109 17.9808 24.8049C18.0194 24.9989 17.9996 25.2 17.9239 25.3827C17.8482 25.5654 17.72 25.7216 17.5556 25.8315C17.3911 25.9414 17.1978 26 17 26C16.7349 25.9997 16.4807 25.8942 16.2932 25.7068C16.1058 25.5193 16.0003 25.2651 16 25Z"
          fill="#865A5A"
        />
        <path
          d="M62 4V2H11C10.8687 2 10.7386 2.02586 10.6173 2.07611C10.496 2.12637 10.3857 2.20003 10.2929 2.29288C10.2 2.38574 10.1264 2.49598 10.0761 2.61731C10.0259 2.73864 10 2.86868 10 3V11C10 11.1313 10.0259 11.2614 10.0761 11.3827C10.1264 11.504 10.2 11.6143 10.2929 11.7071C10.3857 11.8 10.496 11.8736 10.6173 11.9239C10.7386 11.9741 10.8687 12 11 12H14.0445C14.2002 13.2211 14.7474 13.9501 15.2018 14.5522C15.6652 15.1662 16 15.6098 16 16.5998C16.0179 17.0996 15.8922 17.594 15.6379 18.0246C12.8281 18.1862 10.1376 19.2151 7.93701 20.9697C7.3326 21.4533 6.84468 22.0666 6.50938 22.7642C6.17408 23.4619 5.99999 24.226 6 25C6.00001 25.7741 6.17411 26.5382 6.50942 27.2358C6.84472 27.9335 7.33265 28.5467 7.93707 29.0303C9.71099 30.4453 11.8107 31.394 14.045 31.7901C14.0148 32.025 13.9998 32.2617 14 32.4985C14 34.1585 14.6664 35.0414 15.2018 35.7509C15.6652 36.3649 16 36.8085 16 37.7985C16 37.868 15.9972 37.9342 15.9938 38H3C2.86868 38 2.73864 38.0259 2.61731 38.0761C2.49598 38.1264 2.38574 38.2 2.29288 38.2929C2.20003 38.3857 2.12637 38.496 2.07611 38.6173C2.02586 38.7386 2 38.8687 2 39V43C2.00464 46.8037 3.45236 50.464 6.05086 53.2417C8.64936 56.0195 12.2051 57.7078 16 57.9659V61C16 61.1313 16.0259 61.2614 16.0761 61.3827C16.1264 61.504 16.2 61.6143 16.2929 61.7071C16.3857 61.8 16.496 61.8736 16.6173 61.9239C16.7386 61.9741 16.8687 62 17 62H45C45.1313 62 45.2614 61.9741 45.3827 61.9239C45.504 61.8736 45.6143 61.8 45.7071 61.7071C45.8 61.6143 45.8736 61.504 45.9239 61.3827C45.9741 61.2614 46 61.1313 46 61V57.9659C49.7949 57.7078 53.3506 56.0195 55.9491 53.2417C58.5476 50.464 59.9954 46.8037 60 43V39C60 38.8687 59.9741 38.7386 59.9239 38.6173C59.8736 38.496 59.8 38.3857 59.7071 38.2929C59.6143 38.2 59.504 38.1264 59.3827 38.0761C59.2614 38.0259 59.1313 38 59 38H52.8914C53.3964 37.7974 53.872 37.6474 54.3572 37.4946C55.8052 37.0387 57.3026 36.5673 59.6097 34.7926C59.7996 34.6465 59.9307 34.4372 59.9793 34.2026C60.0278 33.968 59.9905 33.7237 59.8741 33.5143L54.8741 24.5143C54.8043 24.3886 54.7081 24.2794 54.5922 24.1942C54.4763 24.1091 54.3434 24.0499 54.2026 24.0207C54.0617 23.9916 53.9162 23.9932 53.776 24.0254C53.6359 24.0576 53.5043 24.1197 53.3903 24.2074C51.364 25.766 50.0975 26.1647 48.7567 26.5869C47.3887 27.0176 45.9763 27.4627 43.8792 29.0068C42.0549 27.992 39.9058 27.7321 37.8922 28.2827C37.9657 27.9261 38.0019 27.5628 38 27.1987C38.0028 26.7942 37.958 26.3907 37.8666 25.9966C37.9112 25.9973 37.9553 26 38 26C39.6255 26.001 41.2128 25.5066 42.5501 24.5825C43.8874 23.6584 44.9112 22.3486 45.4851 20.8278C46.0591 19.3069 46.1557 17.6473 45.7623 16.0701C45.3689 14.4929 44.504 13.0731 43.283 12H62V10H36.2922C36.4406 9.73763 36.6098 9.48759 36.7982 9.25232C37.1096 8.86637 37.3761 8.44623 37.5926 8H62V6H12V4H62ZM30.2018 14.5522C30.335 14.7287 30.4559 14.8918 30.5631 15.0575C30.1907 15.9938 29.9996 16.9923 30 18C30.0002 18.2857 30.0157 18.5712 30.0463 18.8552C29.5457 19.532 29 20.3972 29 21.8995C29 23.5594 29.6664 24.4422 30.2018 25.1516C30.6652 25.7654 31 26.209 31 27.1987C31 28.1887 30.6652 28.6323 30.2018 29.2462C29.6664 29.9556 29 30.8386 29 32.4985C29 34.1585 29.6664 35.0414 30.2018 35.7509C30.6652 36.3649 31 36.8085 31 37.7985C31 37.868 30.9972 37.9342 30.9938 38H27.9933C27.9956 37.9328 28 37.8674 28 37.7985C28 36.1385 27.3336 35.2556 26.7982 34.5461C26.3348 33.9321 26 33.4885 26 32.4985C26 31.5086 26.3348 31.065 26.7982 30.451C27.3336 29.7416 28 28.8587 28 27.1987C28 25.5389 27.3336 24.6561 26.7982 23.9467C26.3348 23.3328 26 22.8893 26 21.8995C26 20.9096 26.3348 20.466 26.7982 19.8521C27.3336 19.1426 28 18.2597 28 16.5998C28 14.9398 27.3336 14.0569 26.7982 13.3474C26.3348 12.7335 26 12.2899 26 11.2999C26 10.3099 26.3348 9.86633 26.7982 9.25238C27.1096 8.86641 27.3761 8.44625 27.5926 8H30.2375C30.2255 8.01596 30.2141 8.03128 30.2018 8.04749C29.6664 8.75696 29 9.63989 29 11.2999C29 12.9598 29.6664 13.8428 30.2018 14.5522ZM25.2018 14.5522C25.6652 15.1662 26 15.6098 26 16.5998C26 17.5897 25.6652 18.0333 25.2018 18.6472C24.6664 19.3566 24 20.2396 24 21.8995C24 23.5594 24.6664 24.4422 25.2018 25.1516C25.6652 25.7654 26 26.209 26 27.1987C26 28.1887 25.6652 28.6323 25.2018 29.2462C24.6664 29.9556 24 30.8386 24 32.4985C24 34.1585 24.6664 35.0414 25.2018 35.7509C25.6652 36.3649 26 36.8085 26 37.7985C26 37.868 25.9972 37.9342 25.9938 38H22.9933C22.9956 37.9328 23 37.8674 23 37.7985C23 36.1385 22.3336 35.2556 21.7982 34.5461C21.3348 33.9321 21 33.4885 21 32.4985C21 31.5086 21.3348 31.065 21.7982 30.4511C22.1045 30.0699 22.3677 29.6559 22.583 29.2168C23.6279 27.8375 24.1235 26.1194 23.9738 24.3954C23.8241 22.6715 23.0398 21.0645 21.7727 19.886C21.7813 19.8746 21.7895 19.8635 21.7982 19.8521C22.3336 19.1427 23 18.2598 23 16.5998C23 14.9398 22.3336 14.0569 21.7982 13.3474C21.3348 12.7335 21 12.2899 21 11.2999C21 10.3099 21.3348 9.86633 21.7982 9.25232C22.1096 8.86637 22.3761 8.44622 22.5925 8H25.2376C25.2255 8.01598 25.2141 8.03132 25.2018 8.04755C24.6664 8.75696 24 9.63989 24 11.2999C24 12.9598 24.6664 13.8428 25.2018 14.5522V14.5522ZM20.2018 14.5522C20.6652 15.1662 21 15.6098 21 16.5998C21 17.5897 20.6652 18.0333 20.2018 18.6473C20.1788 18.6778 20.1545 18.7105 20.1308 18.742C19.4006 18.3749 18.6112 18.1396 17.7991 18.047C17.9372 17.5772 18.0049 17.0895 18 16.5998C18 14.9398 17.3336 14.0569 16.7982 13.3474C16.3348 12.7335 16 12.2899 16 11.2999C16 10.3099 16.3348 9.86633 16.7982 9.25238C17.1096 8.86641 17.3761 8.44625 17.5926 8H20.2375C20.2255 8.01598 20.214 8.03132 20.2018 8.04755C19.6664 8.75696 19 9.63989 19 11.2999C19 12.9598 19.6664 13.8428 20.2018 14.5522V14.5522ZM8 25C7.99888 24.5257 8.10502 24.0573 8.31048 23.6298C8.51594 23.2024 8.8154 22.8269 9.18646 22.5315C11.2365 20.8957 13.7805 20.0033 16.4031 20H17C18.3261 20 19.5979 20.5268 20.5355 21.4645C21.4732 22.4021 22 23.6739 22 25C22 26.3261 21.4732 27.5979 20.5355 28.5355C19.5979 29.4732 18.3261 30 17 30H16.4031C13.7805 29.9967 11.2364 29.1043 9.1864 27.4685C8.81536 27.1731 8.51591 26.7976 8.31046 26.3701C8.10501 25.9427 7.99888 25.4743 8 25ZM16.7982 34.5461C16.3348 33.9321 16 33.4885 16 32.4985C15.9993 32.3291 16.011 32.1598 16.0351 31.9921C16.1577 31.9955 16.2803 32 16.4031 32H17C17.6979 32.0004 18.3918 31.8961 19.0587 31.6906C19.0191 31.9581 18.9994 32.2281 19 32.4985C19 34.1585 19.6664 35.0414 20.2018 35.7509C20.6652 36.3649 21 36.8085 21 37.7985C21 37.868 20.9972 37.9342 20.9938 38H17.9933C17.9956 37.9328 18 37.8674 18 37.7985C18 36.1385 17.3336 35.2556 16.7982 34.5461ZM5.7657 49.5348C5.12123 48.432 4.64389 47.2397 4.34925 45.9968C4.38981 45.9978 4.42972 46 4.47088 46C5.99261 46 6.78522 45.309 7.42212 44.7537C7.94275 44.2999 8.28668 44 9.05896 44C9.83124 44 10.1752 44.2999 10.6957 44.7537C11.3326 45.309 12.1252 46 13.6468 46C15.1685 46 15.9611 45.309 16.598 44.7538C17.1186 44.2999 17.4625 44 18.2348 44C19.0071 44 19.351 44.2999 19.8716 44.7538C20.5085 45.309 21.3011 46 22.8228 46C24.3446 46 25.1373 45.309 25.7741 44.7538C26.2948 44.2999 26.6388 44 27.4111 44C28.1835 44 28.5275 44.2999 29.0482 44.7538C29.6851 45.309 30.4777 46 31.9995 46C33.5211 46 34.3137 45.309 34.9505 44.7537C35.4711 44.2999 35.815 44 36.5872 44C37.3594 44 37.7034 44.2999 38.2239 44.7537C38.8608 45.309 39.6533 46 41.175 46C42.6968 46 43.4894 45.309 44.1263 44.7538C44.647 44.2999 44.991 44 45.7634 44C46.5358 44 46.8798 44.2999 47.4005 44.7538C48.0374 45.309 48.83 46 50.3517 46C51.8736 46 52.6663 45.309 53.3032 44.7538C53.8239 44.2999 54.168 44 54.9404 44C55.7128 44 56.0568 44.2999 56.5775 44.7538C56.928 45.0813 57.3187 45.3628 57.7402 45.5917C57.54 46.5764 57.2253 47.5344 56.8026 48.446C56.2294 48.1428 55.5888 47.9894 54.9404 48C53.4186 48 52.6259 48.691 51.989 49.2462C51.4683 49.7001 51.1243 50 50.3517 50C49.5793 50 49.2354 49.7001 48.7147 49.2462C48.0778 48.691 47.2852 48 45.7634 48C44.2416 48 43.449 48.691 42.8121 49.2462C42.2914 49.7001 41.9474 50 41.175 50C40.4028 50 40.0588 49.7001 39.5383 49.2463C38.9014 48.691 38.1089 48 36.5872 48C35.0656 48 34.2729 48.691 33.6362 49.2463C33.1156 49.7001 32.7717 50 31.9995 50C31.2271 50 30.8831 49.7001 30.3624 49.2462C29.7255 48.691 28.9329 48 27.4111 48C25.8894 48 25.0967 48.691 24.4599 49.2462C23.9392 49.7001 23.5952 50 22.8228 50C22.0505 50 21.7066 49.7001 21.186 49.2462C20.5491 48.691 19.7565 48 18.2348 48C16.7131 48 15.9205 48.691 15.2836 49.2463C14.7631 49.7001 14.4191 50 13.6468 50C12.8746 50 12.5306 49.7001 12.0101 49.2463C11.3732 48.691 10.5806 48 9.05896 48C7.53723 48 6.74463 48.691 6.10779 49.2462C5.98804 49.3506 5.87639 49.447 5.7657 49.5348ZM44 60H18V58H44V60ZM45 56H17C15.0647 56.0002 13.1538 55.5676 11.4072 54.734C9.66067 53.9003 8.12262 52.6866 6.90568 51.1818C7.09597 51.0376 7.26571 50.8901 7.42205 50.7538C7.94275 50.2999 8.28668 50 9.05896 50C9.83124 50 10.1751 50.2999 10.6957 50.7537C11.3326 51.309 12.1252 52 13.6468 52C15.1685 52 15.9611 51.309 16.598 50.7538C17.1186 50.2999 17.4625 50 18.2348 50C19.0071 50 19.351 50.2999 19.8716 50.7538C20.5085 51.309 21.3011 52 22.8228 52C24.3446 52 25.1373 51.309 25.7741 50.7538C26.2948 50.2999 26.6388 50 27.4111 50C28.1835 50 28.5275 50.2999 29.0482 50.7538C29.6851 51.309 30.4777 52 31.9995 52C33.5211 52 34.3137 51.309 34.9505 50.7537C35.4711 50.2999 35.815 50 36.5872 50C37.3594 50 37.7034 50.2999 38.2239 50.7537C38.8608 51.309 39.6533 52 41.175 52C42.6968 52 43.4894 51.309 44.1263 50.7538C44.647 50.2999 44.991 50 45.7634 50C46.5358 50 46.8798 50.2999 47.4005 50.7538C48.0374 51.309 48.83 52 50.3517 52C51.8736 52 52.6663 51.309 53.3032 50.7538C53.8239 50.2999 54.168 50 54.9404 50C55.2463 49.99 55.5499 50.0556 55.8243 50.1912C54.6389 51.9768 53.0302 53.4416 51.1417 54.4551C49.2532 55.4685 47.1433 55.9993 45 56ZM58 43C58 43.1125 57.9986 43.2246 57.9957 43.3364C57.9615 43.3068 57.9272 43.2771 57.8918 43.2462C57.2549 42.691 56.4622 42 54.9404 42C53.4186 42 52.6259 42.691 51.989 43.2462C51.4683 43.7001 51.1243 44 50.3517 44C49.5793 44 49.2354 43.7001 48.7147 43.2462C48.0778 42.691 47.2852 42 45.7634 42C44.2416 42 43.449 42.691 42.8121 43.2462C42.2914 43.7001 41.9474 44 41.175 44C40.4028 44 40.0588 43.7001 39.5383 43.2463C38.9014 42.691 38.1089 42 36.5872 42C35.0656 42 34.2729 42.691 33.6362 43.2463C33.1156 43.7001 32.7717 44 31.9995 44C31.2271 44 30.8831 43.7001 30.3624 43.2462C29.7256 42.691 28.9329 42 27.4111 42C25.8894 42 25.0967 42.691 24.4599 43.2462C23.9392 43.7001 23.5952 44 22.8228 44C22.0505 44 21.7066 43.7001 21.186 43.2462C20.5491 42.691 19.7565 42 18.2348 42C16.7131 42 15.9205 42.691 15.2836 43.2463C14.763 43.7001 14.4191 44 13.6468 44C12.8746 44 12.5307 43.7001 12.0101 43.2463C11.3732 42.691 10.5806 42 9.05896 42C7.53723 42 6.74469 42.691 6.10779 43.2462C5.58716 43.7001 5.24323 44 4.47089 44C4.32489 44.0009 4.17913 43.9879 4.03559 43.9612C4.01228 43.6437 4.00042 43.3233 4 43V40H58V43ZM38 36C38 35.4696 38.2107 34.9609 38.5858 34.5858C38.9609 34.2107 39.4696 34 40 34C40.5304 34 41.0391 34.2107 41.4142 34.5858C41.7893 34.9609 42 35.4696 42 36V38H40C39.4698 37.9994 38.9614 37.7885 38.5865 37.4135C38.2115 37.0386 38.0006 36.5302 38 36ZM49.3572 28.4946C50.5998 28.1034 51.8787 27.7007 53.6744 26.473L57.6984 33.7161C56.0618 34.8612 54.9381 35.215 53.7566 35.5869C52.3912 36.0168 50.9808 36.4614 48.8886 38H47.7477C48.0996 36.6325 48.0831 35.1963 47.7 33.8372C47.3169 32.4781 46.5808 31.2447 45.5666 30.2623C47.121 29.2006 48.2129 28.8549 49.3572 28.4946ZM46 36C46.0009 36.6813 45.8851 37.3578 45.6577 38H44V36C44.0001 35.1091 43.7028 34.2437 43.1552 33.541C42.6076 32.8383 41.841 32.3386 40.9771 32.121C40.1132 31.9035 39.2014 31.9807 38.3863 32.3403C37.5712 32.6999 36.8995 33.3213 36.4778 34.1061C36.1456 33.6376 35.9776 33.0724 36 32.4985C35.9882 32.1005 36.0643 31.7047 36.2229 31.3394C37.1033 30.6255 38.1679 30.1756 39.2934 30.0419C40.419 29.9082 41.5594 30.096 42.5825 30.5838C43.6056 31.0715 44.4697 31.8391 45.0745 32.7977C45.6793 33.7562 46.0002 34.8666 46 36ZM36 27.1987C36 28.1887 35.6652 28.6323 35.2018 29.2462C34.6664 29.9556 34 30.8386 34 32.4985C34 34.1585 34.6664 35.0414 35.2018 35.7509C35.6652 36.3649 36 36.8085 36 37.7985C36 37.868 35.9972 37.9342 35.9938 38H32.9933C32.9955 37.9328 33 37.8674 33 37.7985C33 36.1385 32.3336 35.2556 31.7982 34.5461C31.3348 33.9321 31 33.4885 31 32.4985C31 31.5086 31.3348 31.065 31.7982 30.451C32.3336 29.7416 33 28.8587 33 27.1987C33 25.5389 32.3336 24.6561 31.7982 23.9467C31.3348 23.3328 31 22.8893 31 21.8995C31 21.8889 31.0006 21.8792 31.0007 21.8687C31.9855 23.6431 33.605 24.9794 35.5341 25.6094C35.858 26.0745 36.0215 26.6324 36 27.1987ZM44 18C44.0001 18.8505 43.819 19.6914 43.4687 20.4664C43.1183 21.2414 42.6068 21.9329 41.9683 22.4948C41.989 22.3307 41.9996 22.1654 42 22V18C42 17.2089 41.7654 16.4355 41.3259 15.7777C40.8864 15.1199 40.2616 14.6072 39.5307 14.3045C38.7998 14.0017 37.9956 13.9225 37.2196 14.0769C36.4437 14.2312 35.731 14.6122 35.1716 15.1716C34.6122 15.731 34.2312 16.4437 34.0769 17.2196C33.9225 17.9956 34.0017 18.7998 34.3045 19.5307C34.6072 20.2616 35.1199 20.8864 35.7777 21.3259C36.4355 21.7654 37.2089 22 38 22V20C37.6044 20 37.2178 19.8827 36.8889 19.6629C36.56 19.4432 36.3036 19.1308 36.1522 18.7654C36.0009 18.3999 35.9613 17.9978 36.0384 17.6098C36.1156 17.2219 36.3061 16.8655 36.5858 16.5858C36.8655 16.3061 37.2219 16.1156 37.6098 16.0384C37.9978 15.9613 38.3999 16.0009 38.7654 16.1522C39.1308 16.3036 39.4432 16.56 39.6629 16.8889C39.8827 17.2178 40 17.6044 40 18V22C39.9994 22.5302 39.7885 23.0386 39.4135 23.4135C39.0386 23.7885 38.5302 23.9994 38 24C36.8133 24 35.6533 23.6481 34.6666 22.9888C33.6799 22.3295 32.9108 21.3925 32.4567 20.2961C32.0026 19.1997 31.8838 17.9933 32.1153 16.8295C32.3468 15.6656 32.9182 14.5965 33.7574 13.7574C34.5965 12.9182 35.6656 12.3468 36.8295 12.1153C37.9933 11.8838 39.1997 12.0026 40.2961 12.4567C41.3925 12.9108 42.3295 13.6799 42.9888 14.6666C43.6481 15.6533 44 16.8133 44 18ZM35.2018 8.04749C34.6915 8.72368 34.0636 9.55904 34.0057 11.073C33.0912 11.6025 32.2898 12.3068 31.6471 13.1457C31.2639 12.6268 31 12.1783 31 11.2999C31 10.3099 31.3348 9.86627 31.7982 9.25232C32.1096 8.86634 32.3761 8.44621 32.5925 8H35.2375C35.2255 8.01596 35.2141 8.03128 35.2018 8.04749ZM15.2376 8C15.2255 8.01598 15.2141 8.03132 15.2018 8.04755C14.8271 8.54404 14.389 9.12664 14.1593 10H12V8H15.2376Z"
          fill="#865A5A"
        />
      </svg>
    </Container>
  );
};

const Container = styled.div`
  width: 2rem;
  margin-right: 50%;
`;
