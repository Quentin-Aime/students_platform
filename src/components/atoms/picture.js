import React from "react";
import styled from "styled-components";

export const Picture = ({ imgUrl }) => {
  return (
    <PictureContainer>
      <img src={imgUrl} alt="background" />
    </PictureContainer>
  );
};

const PictureContainer = styled.div`
  max-width: 800px;
  max-height: 800px;
  img {
    width: 100%;
  }
`;
