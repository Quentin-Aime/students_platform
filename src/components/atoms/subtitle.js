import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const SubTitleStyle = styled.h3`
  font-family: "Abril Fatface", cursive;
  font-size: 80px;
  width: 450px;
  line-height: 80px;
  color: #865a5a;
  text-align: ${(props) => (props.alignment ? props.alignment : "left")};
`;

export const SubTitle = ({ content, alignment }) => (
  <SubTitleStyle alignment={alignment}>{content}</SubTitleStyle>
);

SubTitle.displayName = "Subtitle";

SubTitle.defaultProps = {
  alignment: "center",
};

SubTitle.propTypes = {
  alignment: PropTypes.oneOf(["left", "center", "right"]),
  content: PropTypes.string.isRequired,
};
