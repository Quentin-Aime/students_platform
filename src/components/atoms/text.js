import React from "react";
import styled from "styled-components";

export const Text = ({ children, style }) => {
  return <TextContainer style={style}>{children}</TextContainer>;
};

const TextContainer = styled.div`
  font-family: "Poppins";
  line-height: 1.8rem;
  max-width: 100%;
  text-align: left;
`;
