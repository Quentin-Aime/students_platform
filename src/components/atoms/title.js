import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const PageTitleStyle = styled.h1`
  font-family: "Abril Fatface", cursive;
  font-weight: 600;
  font-size: 18px;
  line-height: 80px;
  color: #865a5a;
  text-align: left;
`;

const TitleStyle = styled.h2`
  font-family: "Abril Fatface", cursive;
  font-weight: 600;
  font-size: 18px;
  line-height: 80px;
  color: #865a5a;
  text-align: center;
`;

export const Title = ({ content }) => <TitleStyle>{content}</TitleStyle>;
export const PageTitle = ({ content }) => {
  return <PageTitleStyle>{content}</PageTitleStyle>;
};

/* Will show the right 'tag' within documentation */
PageTitle.displayName = "Page Title";
Title.displayName = "Title";

Title.propTypes = {
  content: PropTypes.element.isRequired,
};

PageTitle.propTypes = {
  content: PropTypes.element.isRequired,
};
