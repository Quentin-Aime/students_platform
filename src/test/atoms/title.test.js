import React from "react";
import { create } from "react-test-renderer";
import { Title } from "../../components/atoms/title";

describe("Title component", () => {
  test("Matches the snapshot", () => {
    const title = create(<Title content={"test"} />);
    expect(title.toJSON()).toMatchSnapshot();
  });
});
