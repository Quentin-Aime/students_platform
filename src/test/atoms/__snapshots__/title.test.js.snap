// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component Matches the snapshot 1`] = `
<h2
  className="sc-AxirZ jOuzpb"
>
  test
</h2>
`;
